src/index.js

```diff
@@ -1,260 +1,308 @@
+                customElement: self,
+                shadowRoot: self.shadowRoot,
+                data() {
+                    return {
+                        props: {},
+                        slotChildren: [],
+                        attrs: getNodeAttributes(self, hyphenatedPropsList, true)
+                    }
+                },
+                render(h) {
+                    return h(Component, {
+                        ref: 'inner',
+                        props: this.props,
+                        attrs: getNodeAttributes(self, hyphenatedPropsList, true)
+                    }, this.slotChildren)
+                }
+            })
+            return wrapper
+        }
 
-    connectedCallback () {
-      if (!this._wrapper) {
-        this._wrapper = this._createWrapper()
-      }
-      const wrapper = this._wrapper
-      if (!wrapper._isMounted) {
-        // initialize attributes
-        const syncInitialProperties = () => {
-          wrapper.props = getInitialProps(camelizedPropsList, wrapper.props)
-          hyphenatedPropsList.forEach(key => {
-            syncProperty(this, key, true)
-          })
+        get vueComponent() {
+            return this._wrapper.$refs.inner
         }
 
-        if (isInitialized) {
-          syncInitialProperties()
-        } else {
-          // async & unresolved
-          Component().then(resolved => {
-            if (resolved.__esModule || resolved[Symbol.toStringTag] === 'Module') {
-              resolved = resolved.default
+        connectedCallback() {
+            if (!this._wrapper) {
+                this._wrapper = this._createWrapper()
+            }
+            const wrapper = this._wrapper
+            if (!wrapper._isMounted) {
+                // initialize attributes
+                const syncInitialProperties = () => {
+                    wrapper.props = getInitialProps(camelizedPropsList, wrapper.props)
+                    hyphenatedPropsList.forEach(key => {
+                        syncProperty(this, key, true)
+                    })
+                }
+
+                if (isInitialized) {
+                    syncInitialProperties()
+                } else {
+                    // async & unresolved
+                    Component().then(resolved => {
+                        if (resolved.__esModule || resolved[Symbol.toStringTag] === 'Module') {
+                            resolved = resolved.default
+                        }
+                        initialize(resolved)
+                        syncInitialProperties()
+                    })
+                }
+                // initialize children
+                wrapper.slotChildren = Object.freeze(toVNodes(
+                    wrapper.$createElement,
+                    this.childNodes
+                ));
+                this._createObserver();
+                wrapper.$mount();
+                this.shadowRoot.appendChild(wrapper.$el);
+            } else {
```

Comment:
I know this isn't your code but I think this would be more readable if you reversed the if condition above and then moved this block above and returned ie. flattening the structure a bit.